autonomic-administration-plugin/src/main/java/br/com/autonomiccs/autonomic/administration/plugin/AdministrationAgent.java

```diff
@@ -276,58 +286,59 @@ private void shutdownIdleHosts(ClusterVO cluster, ClusterAdministrationHeuristic
      * finally it tries every mapped migration with
      * {@link #migrateVM(VMInstanceVO, HostVO)} method.
      */
-    private void mapAndExecuteVMsMigrations(ClusterVO cluster, ClusterAdministrationHeuristicAlgorithm administrationAlgorithm) {
+    protected void mapAndExecuteVMsMigrations(ClusterVO cluster, ClusterAdministrationHeuristicAlgorithm administrationAlgorithm) {
         List<HostResources> hosts = getClusterUpHosts(cluster);
         long clusterId = cluster.getId();
 
         if (CollectionUtils.isEmpty(hosts) || hosts.size() == 1) {
-            logger.debug(String.format("No hosts or single host on cluster[%d], no need to migrate anything.", clusterId));
+            logger.debug(String.format("No hosts or single host on cluster[id=%d], no need to migrate virtual machines.", clusterId));
             return;
         }
 
-        logger.info(String.format("Starting ranking hosts for cluster[%d] administration.", clusterId));
+        logger.info(String.format("Starting ranking hosts for cluster[id=%d] administration.", clusterId));
         List<HostResources> rankedHosts = administrationAlgorithm.rankHosts(hosts);
-        logger.info(String.format("Starting mapping VMs for cluster[%d] administration.", clusterId));
+        logger.info(String.format("Starting mapping VMs for cluster[id=%d] administration.", clusterId));
         Map<Long, HostResources> migrations = administrationAlgorithm.mapVMsToHost(rankedHosts);
-        logger.info(String.format("Starting migrating VMs for cluster[%d] administration.", clusterId));
+        logger.info(String.format("Starting migrating VMs for cluster[id=%d] administration.", clusterId));
 
         for (Entry<Long, HostResources> mappedMig : migrations.entrySet()) {
             Long vmId = mappedMig.getKey();
             HostVO targetHost = hostService.findHostById(mappedMig.getValue().getHostId());
             migrateVM(virtualMachineService.searchVmInstanceById(vmId), targetHost);
         }
-        logger.info(String.format("Migration of VMs for cluster[%d] administration was finished.", clusterId));
+        logger.info(String.format("Migration of VMs for cluster[id=%d] administration was finished.", clusterId));
 
-        logger.info(String.format("Waiting few seconds after VMs migration for cluster[%d] consolidation (if desired)", clusterId));
-        threadUtils.sleepThread(10);// TODO do we need this?
+        logger.info(String.format("Waiting few [%d] seconds after VMs migration for cluster[id=%d].", TIME_TO_WAIT_FOR_MIGRATIONS, clusterId));
+        threadUtils.sleepThread(TIME_TO_WAIT_FOR_MIGRATIONS);//TODO review
```

Comment:
Why is this sleep necessary? If it is for waiting until the actual migration happens, consider creating a thread for migration and issuing a JOIN instead of sleep.